src/services/db/RepoService.ts

```diff
@@ -590,64 +569,13 @@ export default class RepoService extends GitHubService {
       this.gitFileSystemService.push(sessionData.siteName, BRANCH_REF)
       return { newSha: result.value }
     }
-
-    const gitTree = await super.getTree(sessionData, githubSessionData, {
-      isRecursive: true,
-    })
-    const newGitTree: any[] = []
-    const isMovingDirectory =
-      gitTree.find((item: any) => item.path === oldPath)?.type === "tree" ||
-      false
-
-    gitTree.forEach((item: any) => {
-      if (isMovingDirectory) {
-        if (item.path === newPath && item.type === "tree") {
-          throw new ConflictError("Target directory already exists")
-        } else if (item.path === oldPath && item.type === "tree") {
-          // Rename old subdirectory to new name
-          newGitTree.push({
-            ...item,
-            path: newPath,
-          })
-        } else if (
-          item.path.startsWith(`${oldPath}/`) &&
-          item.type !== "tree"
-        ) {
-          // Delete old files
-          newGitTree.push({
-            ...item,
-            sha: null,
-          })
-        }
-      } else if (item.path === newPath && item.type !== "tree") {
-        throw new ConflictError("Target file already exists")
-      } else if (item.path === oldPath && item.type !== "tree") {
-        // Add file to new directory
-        newGitTree.push({
-          ...item,
-          path: newPath,
-        })
-        // Delete old file
-        newGitTree.push({
-          ...item,
-          sha: null,
-        })
-      }
-    })
-
-    const newCommitSha = await super.updateTree(
+    return this.commitServiceGitHub.renameSinglePath(
```

Comment:
didn't look into the method `renameSinglePath`, assuming is just a copy paste over

Reply:
this is the case